app/index.tsx

```diff
@@ -78,163 +78,164 @@ const parseDeepLinking = (url: string) => {
 	return null;
 };
 
-export default class Root extends React.Component<{}, IState> {
-	private listenerTimeout!: any;
-	private dimensionsListener?: EmitterSubscription;
+const Root: React.FC = () => {
+	const [dimensions, setDimensions] = useState<IDimensions>(Dimensions.get('window'));
+	const [state, setState] = useState<{ themePreferences: IThemePreference; theme: TSupportedThemes }>({
+		themePreferences: initialTheme(),
+		theme: getTheme(initialTheme())
+	});
+	const initTheme = initialTheme();
+	const [themeState, setThemeState] = useState<IState>({
+		...state,
+		width: dimensions.width,
+		height: dimensions.height,
+		scale: dimensions.scale,
+		fontScale: dimensions.fontScale
+	});
+
+	useEffect(() => {
+		// init function
```

Comment:
Remove unnecessary comments like this one.
ChatGPT?

Reply:
nope i added it on my own 
while moving i used 2 files while putting all relevent code in the second file i added some comments
yes i used chatGPT but for understanding whats going on not copied the code from it